### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by artyomwhite**
[DEMO LINK](https://artyomwhite.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,48 @@
...
+
+        case Status.ACTIVE:
+          return !todo.completed;
+
+        case Status.COMPLETED:
+          return todo.completed;
+
+        default:
+          throw new Error('Unable to add a todo');
+      }
```
Comment (VitaliyBondarenko1982) [inline src/App.tsx:42]: move filtering inside switch and return filtered todos.
Status.All redundant. just return todos in default case.
```suggestion
      switch (todosFilterBy) {
        case Status.ACTIVE:
          return todos.filter(todo => !todo.completed);

        case Status.COMPLETED:
          return todos.filter(todo => todo.completed);

        default:
           return todos;
      }
```

File: src/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        {`${isActiveTodos.length} items left`}
+      </span>
+
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames('filter__link', {
+            selected: selectItem === Status.ALL,
+          })}
+          onClick={() => setSelectItem(Status.ALL)}
```
Comment (VitaliyBondarenko1982) [inline src/components/Footer.tsx:31]: consider to make one handler factory for all filter buttons and call in jss with proper field
```
const handleSelect = (status: Status) => () => setSelectItem(status);
```
```suggestion
          onClick={handleSelect(Status.ALL)}
```

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Good job.
Needs some improvements.

File: src/App.tsx [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+      switch (todosFilterBy) {
+        case Status.ACTIVE:
+          return !todo.completed;
+        case Status.COMPLETED:
+          return todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [todosFilterBy, todos]);
```
Comment (IvanFesenko) [inline src/App.tsx:39]: Not fixed
2023-08-07_16h28_50

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Do not forget move logic into next TODO`s tasks

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
.

**Pull request closed without merging**